backend/dal/Helpers/Extensions/EntityEntryExtensions.cs

```diff
@@ -17,23 +17,12 @@ public static class EntityEntryExtensions
         /// <param name="entry"></param>
         public static void UndoAppCreatedEdits(this EntityEntry entry)
         {
-            if (entry.Entity is IDisableBaseAppEntity entity)
+            if (entry.Entity is IBaseAppEntity)
             {
-                var oCreatedOn = (DateTime)entry.OriginalValues[nameof(entity.AppCreateTimestamp)];
-                if (!oCreatedOn.Equals(entity.AppCreateTimestamp))
-                    entity.AppCreateTimestamp = oCreatedOn;
-
-                var oCreatedBy = (string)entry.OriginalValues[nameof(entity.AppCreateUserid)];
-                if (String.IsNullOrWhiteSpace(entity.AppCreateUserid) || !oCreatedBy.Equals(entity.AppCreateUserid))
-                    entity.AppCreateUserid = oCreatedBy;
-
-                var oCreatedByKey = (Guid?)entry.OriginalValues[nameof(entity.AppCreateUserGuid)];
-                if (!entity.AppCreateUserGuid.HasValue || !oCreatedByKey.Equals(entity.AppCreateUserGuid))
-                    entity.AppCreateUserGuid = oCreatedByKey;
-
-                var oCreatedByDirectory = (string)entry.OriginalValues[nameof(entity.AppCreateUserDirectory)];
-                if (String.IsNullOrWhiteSpace(entity.AppCreateUserDirectory) || !oCreatedByDirectory.Equals(entity.AppCreateUserDirectory))
-                    entity.AppCreateUserDirectory = oCreatedByDirectory;
+                entry.Property(nameof(IBaseAppEntity.AppCreateTimestamp)).IsModified = false;
```

Comment:
For these time stamps to continue not being set the entity needed to be loaded first. Now is just not set to be tracked.

Reply:
I don't think I understand, if the entity is not loaded we would not be modifying the audit fields on it. This code only operates on loaded entities that are either added, modified, or deleted. 

`public override int SaveChanges()
        {
            // get entries that are being Added or Updated
            var modifiedEntries = ChangeTracker.Entries()
                    .Where(x => x.State.IsIn(EntityState.Added, EntityState.Modified, EntityState.Deleted));

            // Default values are provided because depending on the claim source it may or may not have these values.
            var username = _httpContextAccessor.HttpContext.User.GetUsername() ?? "service";
            var key = _httpContextAccessor.HttpContext.User.GetUserKey();
            var directory = _httpContextAccessor.HttpContext.User.GetUserDirectory() ?? "";
            foreach (var entry in modifiedEntries)
            {
                entry.UpdateAppAuditProperties(username, key, directory);
            }

            return base.SaveChanges();
        }`

Reply:
Ef allows performing updates on entities that have not been loaded into context. It is like doing a SQL update without having to do a select first. AFAIK the tracker keeps a tab of what entities are set to be added/modified/deleted and does not need to load said entities.

Reply:
definitely a good correction though, thanks for this.